src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/RazorFormattingTest.cs.cs

```diff
@@ -191,7 +193,130 @@ private void IncrementCount()
         currentCount++;
     }
 }
+",
+fileKind: FileKinds.Legacy);
+        }
+
+        [Fact]
+        public async Task Layout()
```

Comment:
Glorious set of tests. @pranavkm @captainsafia @javiercn any other directive formatting tests we should encompass to ensure we're formatting the "right" way?

Reply:
This set looks pretty good! We might want to add a test for `@implements` for additional coverage and maybe `@preservewhitespace` since it captures the category of single keyword attributes?

Reply:
> maybe `@preservewhitespace` since it captures the category of single keyword attributes?

So `@preservewhitespace` on its own is valid? I thought it might be but the docs weren't entirely clear.

I'll add a couple more in a follow up

Reply:
Nope, `@preservewhitespace` on its own is an error. Still can add tests covering more directives though, can't hurt

Reply:
Ah, my bad. The docs got the best of me too. We document that it needs a value in the migration doc but not in the doc page on directives.